src/Board/List/TasksCards/TasksCard.js

```diff
@@ -0,0 +1,227 @@
+
+  let listsId = listsFilter.map((el) => el.id);
+  let stateTask = useSelector((state) => state.tasks);
+
+  let tasksFilter = stateTask.filter((item) =>
+    listsId.includes(item[0].listId)
+  );
+
+  useEffect(() => {
+    setLocalDataUserTask(stateTask);
+  }, [stateTask]);
+
+  if (
+    !useSelector((state) => state.boards).find((elem) => elem.id === boardId)
+  ) {
+    return <PageNotFound />;
+  }
+  function onDragEnd(result) {
+    const { source, destination } = result;
+    if (!destination) {
+      return;
+    }
+    const sInd = +source.droppableId;
+    const dInd = +destination.droppableId;
+
+    if (sInd === dInd) {
+      const items = reorder(tasksFilter[sInd], source.index, destination.index);
+      const newState = [...tasksFilter];
+      newState[sInd] = items;
+
+      dispatch(setTaskState(newState, listsId));
+    } else {
+      const result = move(
+        tasksFilter[sInd],
+        tasksFilter[dInd],
+        source,
+        destination
+      );
+      const newState = [...tasksFilter];
+      newState[sInd] = result[sInd];
+      let listId = result[dInd][0].listId;
+      newState[dInd] = result[dInd].map(
+        (elem, ind) =>
+          (elem =
+            ind === destination.index
+              ? {
+                  id: elem.id,
+                  name: elem.name,
+                  listId: listId,
+                  description: elem.description,
+                }
+              : elem)
+      );
+      dispatch(
+        setTaskState(
+          newState.filter((group) => group.length),
+          listsId
+        )
+      );
+    }
+  }
+
+  return (
+    <Card className={`${s.totalCard} totalCard`}>
+      <DragDropContext onDragEnd={onDragEnd}>
+        {tasksFilter.map((el, ind) => (
+          <Card.Grid key={`boxList${el[0].listId}`} className={s.card}>
+            <Card
+              className={`${s.tasksHeader} tasksHeader`}
+              key={`listone${el[0].listId}`}
+              title={
+                <div key={`headerCard${el[0].listId}`}>
+                  <List listId={el[0].listId} key={`lists${el[0].listId}`} />
+                  <NewTask
+                    stateTasks={tasksFilter}
+                    listsId={listsId}
+                    listId={el[0].listId}
+                    uuid={uuid()}
+                    key={`newTask${el[0].listId}`}
+                  />
+                </div>
+              }
+            >
+              <Droppable
+                droppableId={`${ind}`}
+                key={`droppable${el[0].listId}`}
+              >
+                {(provided, snapshot) => (
+                  <div
+                    key={`tasks${el[0].listId}`}
+                    ref={provided.innerRef}
+                    style={getListStyle(snapshot.isDraggingOver)}
+                    {...provided.droppableProps}
+                    style={{ width: "100%" }}
+                  >
+                    {el.map((item, index) => (
```

Comment:
There are a lot of nesting. Create separate components/functions to reduce nesting